storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/dataaccess/V4FinalizedRocksDbDao.java

```diff
@@ -63,6 +67,16 @@ public void close() throws Exception {
         .map(ColumnEntry::getValue);
   }
 
+  @Override
+  public Set<SignedBeaconBlock> getNonCanonicalBlocksAtSlot(final UInt64 slot) {
+    Optional<Set<Bytes32>> maybeRoots = db.get(schema.getColumnNonCanonicalRootsBySlot(), slot);
+    return maybeRoots.stream()
+        .flatMap(Collection::stream)
+        .map(root -> db.get(schema.getColumnNonCanonicalBlocksByRoot(), root))
+        .flatMap(Optional::stream)
```

Comment:
nit: I'd be tempted to combine these two lines:
```suggestion
        .flatMap(root -> db.get(schema.getColumnNonCanonicalBlocksByRoot(), root).stream())
```